google-cloud-pubsub/lib/google/cloud/pubsub/subscriber/stream.rb

```diff
@@ -107,6 +108,9 @@ def paused?
           end
 
           def wait!
+            # Wait for all queued callbacks to be processed.
+            @callback_thread_pool.wait_for_termination
```

Comment:
Should this line have a timeout value provided? "When the timeout value is nil the call will block indefinitely."

Ideally a configurable timeout, but elsewhere in this library it is hard-coded to 60.